lib/hydra/works/services/generic_work/move_to_generic_file.rb

```diff
@@ -0,0 +1,34 @@
+module Hydra::Works
+  class MoveGenericFileToGenericWork
+
+    ##
+    # Move a generic file from one generic work to another
+    #
+    # @param [Hydra::Works::GenericWork] :old_parent_work where the generic file currently lives
+    # @param [Hydra::Works::GenericWork] :new_parent_work where the generic file is being moved
+    # @param [Hydra::Works::GenericFile] :child_generic_file being moved
+    #
+    # @return [Hydra::Works::GenericWork] the destination generic work
+    def self.call(old_parent_work, new_parent_work, child_generic_file)
+      raise ArgumentError, "old_parent_work must be a Hydra::GenericWork" unless old_parent_work.is_a?(Hydra::Works::GenericWork)
+      raise ArgumentError, "new_parent_work must be a Hydra::GenericWork" unless new_parent_work.is_a?(Hydra::Works::GenericWork)
+      raise ArgumentError, "child_generic_file must be a Hydra::GenericFile" unless child_generic_file.is_a?(Hydra::Works::GenericFile)
```

Comment:
change these to use the newly added Hydra::Works.generic_work? and .generic_file? methods.